### PR TITLE
Tab: Only make `Tab` an anchor if a `href` is passed

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- render `Tab` as a `<button>` by default, only render as an anchor when an `href` is present

**Why do we need this feature?**

- so we don't always render `<Tab>`s as anchor elements with empty `href`s

**Who is this feature for?**

- everyone!

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: .betterer.results [hunk 1]
```diff
@@ -928,6 +928,10 @@ exports[`better eslint`] = {
       [0, 0, 0, "Unexpected any. Specify a different type.", "4"],
       [0, 0, 0, "Unexpected any. Specify a different type.", "5"]
     ],
+    "packages/grafana-ui/src/components/Tabs/Tab.tsx:5381": [
+      [0, 0, 0, "Do not use any type assertions.", "0"],
+      [0, 0, 0, "Do not use any type assertions.", "1"]
+    ],
```
Comment (joshhunt) [inline .betterer.results:934]: We don't plan on "fixing" these type assertions, so use eslint ignore comments to silence the linter instead.

File: packages/grafana-ui/src/components/Tabs/Tab.tsx [hunk 2]
```diff
@@ -36,23 +39,41 @@ export const Tab = React.forwardRef<HTMLAnchorElement, TabProps>(
...
     );
 
-    const linkClass = cx(tabsStyles.link, active ? tabsStyles.activeStyle : tabsStyles.notActive);
+    const linkClass = cx(clearStyles, tabsStyles.link, active ? tabsStyles.activeStyle : tabsStyles.notActive);
 
-    return (
+    if (href) {
       <div className={tabsStyles.item}>
         <a
-          // in case there is no href '#' is set in order to maintain a11y
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/Tabs/Tab.tsx:44]: lol

File: packages/grafana-ui/src/components/Tabs/Tab.tsx [hunk 3]
```diff
@@ -36,23 +39,43 @@ export const Tab = React.forwardRef<HTMLAnchorElement, TabProps>(
       </>
     );
 
-    const linkClass = cx(tabsStyles.link, active ? tabsStyles.activeStyle : tabsStyles.notActive);
+    const linkClass = cx(clearStyles, tabsStyles.link, active ? tabsStyles.activeStyle : tabsStyles.notActive);
 
-    return (
+    if (href) {
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/Tabs/Tab.tsx:44]: Missing the return statement here.

File: packages/grafana-ui/src/components/Tabs/Tab.tsx [hunk 4]
```diff
@@ -36,23 +39,43 @@ export const Tab = React.forwardRef<HTMLAnchorElement, TabProps>(
...
+          className={linkClass}
+          {...otherProps}
+          onClick={onChangeTab}
+          aria-label={otherProps['aria-label'] || selectors.components.Tab.title(label)}
+          role="tab"
+          type="button"
+          aria-selected={active}
+          // don't think we can avoid the type assertion here :(
+          // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+          ref={ref as React.ForwardedRef<HTMLButtonElement>}
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/Tabs/Tab.tsx:75]: Most of these props are the same as for the anchor, should we extract them into a separate object, so they're not declared twice?

File: packages/grafana-ui/src/components/Tabs/Tab.tsx (hunk 3 above)
Comment (ashharrison90) [reply packages/grafana-ui/src/components/Tabs/Tab.tsx:44]: no one saw anything 🤫

**Review by eledobleefe — APPROVED** [review]
Tested on local and it works 👌

**Review by Clarity-89 — APPROVED** [review]

**Pull request merged**